config/v1/types_cluster_version.go

```diff
@@ -806,6 +844,18 @@ type ConditionalUpdate struct {
 // for not recommending a conditional update.
 // +k8s:deepcopy-gen=true
 type ConditionalUpdateRisk struct {
+	// conditions represents the observations of the conditional update
+	// risk's current status. Known types are:
+	// * Applies, for whether the risk applies to the current cluster.
```

Comment:
In the past Joel made me use CEL to ensure only the valid conditions appear in the list. Not sure if this applies here too.

Reply:
Tried to see an example with `rg '\[\]meta.*.Condition.*json' -B10` and found nothing.

`kubebuilder:validation:Enum` is for the field directly but we need to do it on its `struct`. Something like `kubebuilder:validation:Enum(self.Type)=Applied` would be nice.

```go
// +kubebuilder:validation:Enum={val1,val2,val3}: Restricts a field to a set of allowed values.
```

Reply:
If you wanted to restrict this to only contain specific condition types you would use the `kubebuilder:validation:items:XValidation` marker to write a CEL expression that accesses the `self.type` field of each condition entry and checks that it is one of the correct types.

Reply:
I do not find any existing example. Google guides me with

```go
// +kubebuilder:validation:items:XValidation:rule="has(self.Type) && self.Type == 'Applies'"
```

`code-gen` recognizes it.

Let us see if it works.
I will add an example for it too.

Reply:
Done.

See https://github.com/openshift/api/pull/2360#issuecomment-3046264007